src/main/resources/com/puppycrawl/tools/checkstyle/grammars/javadoc/JavadocLexer.g4

```diff
@@ -37,8 +37,7 @@ import java.util.*;
 
 }
 
-LEADING_ASTERISK : ( (' '|'\t') {_tokenStartCharPositionInLine == 0}? ) (' '|'\t')* '*'
-      | '*' {_tokenStartCharPositionInLine == 0}?
+LEADING_ASTERISK : (' '|'\t')* '*'
```

Comment:
I didn't realize `_tokenStartCharPositionInLine` is a field that came from ANTLR itself.
It is populated via `_tokenStartCharPositionInLine = getInterpreter().getCharPositionInLine();` which is defined at https://github.com/antlr/antlr4/blob/master/runtime/Java/src/org/antlr/v4/runtime/atn/LexerATNSimulator.java#L736-L746 .
It is only reset to 0 on `\n` so this is why the fix with `\r` doesn't work anymore. It doesn't see it as a new line.

I question why we had this special code before if it was never really needed.
@ps-sp If you print the ANTLR tree with only `\r` does it display the line numbers the same as if you were using `\n`?